src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -269,36 +272,35 @@ class JooqBurdenEstimateRepository(
     override fun populateBurdenEstimateSet(setId: Int, groupId: String, touchstoneId: String, scenarioId: String,
                                            estimates: Sequence<BurdenEstimateWithRunId>)
     {
-        // Dereference modelling group IDs
-        val modellingGroup = modellingGroupRepository.getModellingGroup(groupId)
-
-        val responsibilityInfo = getResponsibilityInfo(modellingGroup.id, touchstoneId, scenarioId)
-
-        val set = getBurdenEstimateSetForResponsibility(setId, responsibilityInfo.id)
+        val (set, responsibilityInfo) = getSetFromResponsibilityPath(groupId, touchstoneId, scenarioId, setId)
         val type = set.type.type
 
         if (set.status == BurdenEstimateSetStatus.COMPLETE)
         {
-            throw OperationNotAllowedError("This burden estimate set has been marked as complete." +
+            throw InvalidOperationError("This burden estimate set has been marked as complete." +
                     " You must create a new set if you want to upload any new estimates.")
         }
 
-        if (type == BurdenEstimateSetTypeCode.STOCHASTIC)
-        {
-            stochasticBurdenEstimateWriter.addEstimatesToSet(setId, estimates, responsibilityInfo.disease)
-        }
-        else
-        {
-            centralBurdenEstimateWriter.addEstimatesToSet(setId, estimates, responsibilityInfo.disease)
-        }
-
+        getEstimateWriter(set).addEstimatesToSet(setId, estimates, responsibilityInfo.disease)
         changeBurdenEstimateStatus(setId, BurdenEstimateSetStatus.PARTIAL)
         updateCurrentBurdenEstimateSet(responsibilityInfo.id, setId, type)
     }
 
-    override fun closeBurdenEstimateSet(setId: Int)
+    override fun closeBurdenEstimateSet(setId: Int, groupId: String, touchstoneId: String, scenarioId: String)
     {
-        changeBurdenEstimateStatus(setId, BurdenEstimateSetStatus.COMPLETE)
+        // Check all the IDs match up
+        val (set, _) = getSetFromResponsibilityPath(groupId, touchstoneId, scenarioId, setId)
+        val helper = getEstimateWriter(set)
+        val isEmpty = dsl.fetchAny(helper.table, helper.setField.eq(setId)) == null
```

Comment:
Here I'm using the `BurdenEstimateWriter` to look at the appropriate table, which means maybe we should rename it to `BurdenEstimateHelper`? Or maybe it's fine as is.